### ISSUE TITLE
no-indistinguishable-colors should ignore color names in URL's

### ISSUE THREAD

**Issue by ntwb** [stylelint/stylelint#970]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug where if a URL contains a color name it is incorrectly detected as a property value.

> Which rule, if any, is this issue related to?

`no-indistinguishable-colors`

> What CSS is needed to reproduce this issue?

```css
.a {
	background-image: url(../images/white.png);
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "no-indistinguishable-colors": true,
  }
}
```

> Which version of stylelint are you using?

`5.2.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

grunt-postcss

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
test.css
71:34	⚠  Unexpected almost identical colors: "white" and "#fff" (no-indistinguishable-colors) [stylelint]
```

**Comment by evilebottnawi** [comment]
@ntwb issue can not be repeat, not in test, not in my CSS code with above example.

**Comment by ntwb** [comment]
Added a failing test in #977

**Comment by davidtheclark** [comment]
This is a bug with css-colorguard. We can probably try to patch it in some way on our end, but please file an issue over there, too, when you have a chance.

**Comment by davidtheclark** [comment]
Oh, looks like you did that :)

**Comment by ntwb** [comment]
Thanks, currently with a couple of thousand warnings when using `no-indistinguishable-colors` I'm in no hurry to start using this, happy to wait for upstream fix rather than patching stylelint.

@evilebottnawi Thanks for testing and creating the upstream issue :+1:

**Comment by davidtheclark** [comment]
> currently with a couple of thousand warnings

Wow. Do you think those are mostly false positives?

**Comment by ntwb** [comment]
> 

Nope, legit, just needs an audit, I think, I haven't looked to closely really either just yet, but will do.

Here's a [gist](https://gist.github.com/ntwb/aaae281807d1a8c8380f) of the output with the default threshold.

Issues like this below could be looked into, i.e. using an alpha transparency with a fallback for IE8 should probably be ignored, though as IE9 and above supports alpha transparency and I think WordPress is about to drop IE8 support this issue would be a non-issue by the time (I hope) to get to this.
 
```shell
1002:24	⚠  Unexpected almost identical colors: "rgba(0, 0, 0, 0.3)" and "#000" (no-indistinguishable-colors) [stylelint]
```

**Comment by davidtheclark** [comment]
I believe this is just awaiting the next release of css-colorguard, in which this should be resolved.
